### DISCUSSION TITLE
Support SharePoint Permission Groups

### DISCUSSION THREAD

**Discussion by tbithell** [pnp/pnpcore#500] [Ideas]
For those of us that still use sharepoint permission groups and permission levels it'd be real nice to be able to handle that this way since it's a lot easier to write unit tests against pnp core sdk than it is against csom.

**Comment by jansenbe** [comment]
@tbithell : we do support getting the groups from a web via `SiteGroups`, `AssociatedMemberGroup`, `AssociatedOwnerGroup` and `AssociatedVisitorGroup`. These will return a (collection of) `ISharePointGroup` for you to work with. You can configure custom role definitions for a group etc + there are `BreakRoleInheritance` methods on `IList` and `IListItem`.

Documentation for these features is still lacking, but you can find some basic unit tests showing the usage here: https://github.com/pnp/pnpcore/tree/dev/src/sdk/PnP.Core.Test/Security
